### PR TITLE
feat: json-api query helpers

### PR DESCRIPTION
**Pull request by therockstorm**
Summary
===
Add `json-api` query helpers inspired by https://github.com/scoutforpets/bookshelf-jsonapi-params and https://github.com/kideh88/node-jsonapi-query-parser. See the README for example usage.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -33,6 +33,7 @@ jobs:
           # GitHub checks PRs out based on the merge commit; we want the branch HEAD.
           ref: ${{ github.event.pull_request.head.sha }}
           fetch-depth: 0
+      - run: git fetch origin main:main
```
Comment (FelipePhilippCbh) [inline .github/workflows/ci.yml:36]: Is this needed?

File: packages/json-api/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# @clipboard-health/json-api
+
+TODO
```
Comment (FelipePhilippCbh) [inline packages/json-api/README.md:3]: Missing this

File: packages/json-api/src/lib/types.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/**
+ * Filter operators to build complex filter queries.
+ * - gt: Greater than
+ * - ge: Greater than or equal to
+ * - lt: Less than
+ * - le: Less than or equal to
+ * - not: Not equal to
+ */
+type FieldType = "gt" | "ge" | "lt" | "le" | "not";
```
[the remark this answers is not in the document]
Comment (fasidOnGit) [reply packages/json-api/src/lib/types.ts:9]: in JSON:API that would be `,` seperated value. like `/movies?filter[rating]=G,PG-13`

**Review by fasidOnGit — APPROVED** [review]
lgtm

File: .github/workflows/ci.yml (hunk 1 above)
Comment (therockstorm) [reply .github/workflows/ci.yml:36]: Yes, we had to add it to `cbh-core` too.

File: packages/json-api/src/lib/types.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (therockstorm) [reply packages/json-api/src/lib/types.ts]: The comma-separated list seems just as easy to understand and adheres to the spec, so I'll move forward with that.

**Review by FelipePhilippCbh — APPROVED** [review]

**Pull request merged**